pymeasure/display/widgets.py

```diff
@@ -39,11 +39,43 @@
 from .Qt import QtCore, QtGui
 from ..experiment import parameters, Procedure
 from ..experiment.results import Results
+from functools import partial
 
 log = logging.getLogger(__name__)
 log.addHandler(logging.NullHandler())
 
 
+def input_from_parameter(parameter):
```

Comment:
I defined a general function to get the type of Input that corresponds to a given type of Parameter. The code is taken from the InputsWidget and this function is also used in that class.